powertools-cloudformation/src/main/java/software/amazon/lambda/powertools/cloudformation/AbstractCustomResourceHandler.java

```diff
@@ -65,7 +65,12 @@ public final Response handleRequest(CloudFormationCustomResourceEvent event, Con
         } catch (CustomResourceResponseException rse) {
             LOG.error("Unable to generate response. Sending empty failure to {}", responseUrl, rse);
             try {
-                client.send(event, context, Response.failed());
+                // If the customers code throws an exception, Powertools should respond in a way that doesn't
+                // change the CloudFormation resources.
+                // In the case of a Update or Delete, a failure is sent with the existing PhysicalResourceId
+                // indicating no change.
+                // In the case of a Create, null will be set and changed to the Lambda LogStreamName before sending.
+                client.send(event, context, Response.failed(event.getPhysicalResourceId()));
```

Comment:
ID Injection - we're doing this in two places - here, and further down the page-and-also-call-stack (comment added). On the one hand we're trying to keep the interfaces from changing at this point and deferring that to V2, on the other hand and we've got solid test coverage now, on the other, its a bit irky that this piece of fairly important logic is spread about.

Reply:
I agree, closing.